src/business-logic/services/ingested-entity-to-entity-mapper.ts

```diff
@@ -0,0 +1,194 @@
+import { IngestedPiece } from '../../model/entities/ingested-piece'
+import { Piece } from '../../model/entities/piece'
+import { IngestedPart } from '../../model/entities/ingested-part'
+import { Part } from '../../model/entities/part'
+import { PartTimings } from '../../model/value-objects/part-timings'
+import { Exception } from '../../model/exceptions/exception'
+import { ErrorCode } from '../../model/enums/error-code'
+import { IngestedSegment } from '../../model/entities/ingested-segment'
+import { Segment } from '../../model/entities/segment'
+import { IngestedRundown } from '../../model/entities/ingested-rundown'
+import { Rundown, RundownAlreadyActiveProperties } from '../../model/entities/rundown'
+
+export class IngestedEntityToEntityMapper {
+
+  constructor() { }
+
+  public fromIngestedRundown(ingestedRundown: IngestedRundown): Rundown {
+    return new Rundown({
+      id: ingestedRundown.id,
+      name: ingestedRundown.name,
+      showStyleVariantId: ingestedRundown.showStyleVariantId,
+      segments: [],
+      history: [],
+      isRundownActive: false,
+      modifiedAt: ingestedRundown.modifiedAt,
+      baselineTimelineObjects: ingestedRundown.baselineTimelineObjects,
+      timing: ingestedRundown.timings
+    })
+  }
+
+  public updateRundownFromIngestedRundown(rundownToUpdate: Rundown, ingestedRundown: IngestedRundown): Rundown {
```

Comment:
The same goes for some of the other methods in this class.

Reply:
That's intentional. From a consumer, calling this class, point of view we just want to update our entity, so we call an update method.
Because of immutability we have to create a new instance.

Reply:
Fair enough.